feat: use custom domain linkdex.dag.haus

olizilla opened the pull request:
prod lives at `linkdex.dag.haus`
staging is at `staging.linkdex.dag.haus`

...all non-prod stages follow that pattern `${stage}.linkdex.dag.haus`

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

olizilla committed:
feat: use custom domain linkdex.dag.haus

prod lives at `linkdex.dag.haus`
staging is at `staging.linkdex.dag.haus`

...all non-prod stages follow that pattern `${stage}.linkdex.dag.haus`

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

olizilla commented:
linkdex.dag.haus is updated in cloud flare to use route53 nameservers

olizilla committed:
fix: fix domain template string

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

olizilla committed:
feat: show customDomain in output

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>

vasco-santos reviewed (commented):
Changes look good. For consistency with all other projects I would go with `linkdex-staging` like we do for `api-staging.*`, `ipfs-staging.*`, and so on

olizilla commented:
@vasco-santos good point! Alas, that would be difficult to implement here as the dag.haus domain is managed on cloudflare not route53. The trick that makes this work at all is the subdomain `linkdex.dag.haus` is manually delegated to route53 name servers. I could special case staging and delegate both `linkdex.dag.haus` and `linkdex-staging.dag.haus` but I prefer the look of `${stage}.linkdex.dag.haus` and the original reason for the `api-staging` pattern was that it was hard to generate all the necessary SSL certificates for sub-sub-domains, which isn't the case here.

alanshaw commented on stacks/LinkdexStack.ts:
```diff
@@ -1,14 +1,21 @@
     ? new Bucket(stack, 'existing-cars', { cdk: { bucket: s3.Bucket.fromBucketName(stack, 'imported-cars', process.env.BUCKET_NAME) }})
     : new Bucket(stack, 'cars')
 
+  const zone = 'linkdex.dag.haus'
```
```suggestion
  const zone = process.env.DOMAIN || 'linkdex.dag.haus'
```

olizilla replied:
Good point. I think this needs to be tweaked so that we only set the `customDomain` config on the api if we find a env var, so that it can be run on any aws account without having to update config.

olizilla committed:
chore: make HOSTED_ZONE configureable

remove the hardcoded customDomain so it can be configured for environments that need it.

License: MIT
Signed-off-by: Oli Evans <oli@protocol.ai>
